### PR TITLE
Fix missing icons and colorpicker when mounted on suburl

### PR DESCRIPTION
**Pull request by zeripath**
icons and colorpicker are missing when mounted on a suburl. This PR fixes this.

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -182,19 +182,19 @@ export default {
         type: 'asset/source',
       },
       {
-        test: /\.(ttf|woff2?)$/,
+        test: /\.(ttf|woff2?)$/g,
```
Comment (silverwind) [inline webpack.config.js:185]: global flag should not be necessary

Comment (zeripath) [reply webpack.config.js:185]: It is

Comment (silverwind) [inline webpack.config.js]:
```suggestion
        test: /\.(ttf|woff2?)$/i,
```
global flag should not be necessary

File: webpack.config.js [hunk 2]
```diff
@@ -182,19 +182,19 @@ export default {
...
         type: 'asset/resource',
         generator: {
           filename: 'fonts/[name][ext]',
-          publicPath: '/', // required to remove css/ path segment
+          publicPath: '../', // required to remove css/ path segment
         }
       },
       {
-        test: /\.png$/i,
+        test: /\.png$/ig,
```
Comment (silverwind) [inline webpack.config.js]:
```suggestion
        test: /\.png$/i,
```

**Comment by zeripath** [comment]
The global flag is necessary

From try.gitea.io/index.css unminified:

```
@font-face {
    font-family: Icons;
    src: url(themes/default/assets/fonts/icons.eot);
    src: url(themes/default/assets/fonts/icons.eot?#iefix) format("embedded-opentype"),url(/fonts/icons.woff2) format("woff2"),url(themes/default/assets/fonts/icons.woff) format("woff"),url(themes/default/assets/fonts/icons.ttf) format("truetype"),url(themes/default/assets/fonts/icons.svg#icons) format("svg");
    font-style: normal;
    font-weight: 400;
    font-variant: normal;
    text-decoration:inherit;text-transform:none}
```

I think we even need to do the eot ones too

**Comment by silverwind** [comment]
> I think we even need to do the eot ones too

Modern browsers (e.g. those we support) should not hit on EOT/SVG fonts but prefer WOFF.

> The global flag is necessary

Why? global flag is pointless for simple `regex.test` which I think is what webpack would perform. It does not matter if there is more than 1 match.

**Comment by zeripath** [comment]
you're right. Strangely I was sure that the g did the thing I wanted it to do.

Sorry.

**Review by silverwind — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Comment by lafriks** [comment]
please send backport

**Comment by zeripath** [comment]
I don't think this actually does need a backport

**Pull request merged**